exonum/src/runtime/rust.rs

```diff
@@ -98,44 +113,40 @@ impl RuntimeEnvironment for RustRuntime {
         Ok(())
     }
 
-    fn finish_init(&self, _: &mut EnvContext, _: ServiceInstanceId, abort: bool) {
-        if abort {
-            unimplemented!()
-        }
-    }
-
-    fn execute(&self, context: &mut EnvContext, dispatch: DispatchInfo, payload: &[u8]) {
+    fn execute(
+        &self,
+        context: &mut EnvContext,
+        dispatch: CallInfo,
+        payload: &[u8],
+    ) -> Result<(), ExecutionError> {
         let inner = self.inner.read().unwrap();
```

Comment:
It is out of scope of this PR and I wrote this line of code, but I just wanted to mention that maybe it makes more sense to not have locks inside runtime environment so we don't acquire lock every time we execute transaction.

Reply:
Sure. I guess we'll clarify how to do it when we'll start work on `RustRuntime` task :)